### PR TITLE
MachineProducer implementations created, MachineServiceImpl improved

### PR DESCRIPTION
**Pull request by ivanlavrinenko**

### CODE REVIEW

File: src/main/java/core/mate/academy/service/BulldozerProducer.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+import core.mate.academy.model.Bulldozer;
+import java.util.ArrayList;
+import java.util.List;
+
+public class BulldozerProducer implements MachineProducer<Bulldozer> {
+    BulldozerProducer() {
+
+    }
+
```
Comment (koval-kn) [inline src/main/java/core/mate/academy/service/BulldozerProducer.java]:
```suggestion
```

File: src/main/java/core/mate/academy/service/ExcavatorProducer.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+import core.mate.academy.model.Excavator;
+import java.util.ArrayList;
+import java.util.List;
+
+public class ExcavatorProducer implements MachineProducer<Excavator> {
+    ExcavatorProducer() {
+
+    }
+
```
Comment (koval-kn) [inline src/main/java/core/mate/academy/service/ExcavatorProducer.java:11]:
```suggestion
```

File: src/main/java/core/mate/academy/service/MachineServiceImpl.java [hunk 3]
```diff
@@ -1,7 +1,44 @@
...
+        if (type == Bulldozer.class) {
+            BulldozerProducer bulldozerProducer = new BulldozerProducer();
+            return (List<T>) bulldozerProducer.get();
+        }
+        if (type == Truck.class) {
+            TruckProducer truckProducer = new TruckProducer();
+            return (List<T>) truckProducer.get();
+        }
+        if (type == Excavator.class) {
+            ExcavatorProducer excavatorProducer = new ExcavatorProducer();
```
Comment (koval-kn) [inline src/main/java/core/mate/academy/service/MachineServiceImpl.java]: consider making producers as a `private final` class-level field in order not to create their instances each time when we call this method

File: src/main/java/core/mate/academy/service/TruckProducer.java [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.List;
+
+public class TruckProducer implements MachineProducer<Truck> {
+    TruckProducer() {
+
+    }
+
```
Comment (koval-kn) [inline src/main/java/core/mate/academy/service/TruckProducer.java:11]:
```suggestion
```

File: src/main/java/core/mate/academy/service/BulldozerProducer.java
Comment (koval-kn) [inline src/main/java/core/mate/academy/service/BulldozerProducer.java:1]: according to the task, you need to add some fields to the models (bulldozer, excavator, truck)

**Review by koval-kn — CHANGES_REQUESTED** [review]

File: src/main/java/core/mate/academy/service/ExcavatorProducer.java (hunk 2 above)
Comment (Rommelua) [reply src/main/java/core/mate/academy/service/ExcavatorProducer.java:11]: not fixed

**Review by Rommelua — APPROVED** [review]

**Pull request closed without merging**